[BUG]  windows 编译： bug,Caused by: java.io.IOException: Cannot run program "/bin/bash" (in directory "C:\Users\zhou\AppData\Local\Temp\agent-workspace"): CreateProcess error=2, 系统找不到指定的文件。         a

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

Caused by: java.io.IOException: Cannot run program "/bin/bash" (in directory "C:\Users\zhou\AppData\Local\Temp\agent-workspace"): CreateProcess error=2, 系统找不到指定的文件。
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143) ~[na:na]
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073) ~[na:na]

2 在c:\spring-ai-alibaba\examples\chatbot>mvn spring-boot:run

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

```markdown
Spring AI Alibaba version(s):
```

### Debug logs

_No response_

### Anything else?

_No response_

--- a/spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/tools/ShellSessionManager.java
+++ b/spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/tools/ShellSessionManager.java
@@ -574,14 +574,23 @@ public class ShellSessionManager {
 		 */
 		private static List<String> getDefaultShellCommand() {
 			String os = System.getProperty("os.name").toLowerCase();
+			
 			if (os.contains("windows")) {
 				// Windows: use PowerShell for better compatibility and features
 				return Arrays.asList("powershell.exe");
+			} else if (os.contains("mac")) {
+				// macOS: prefer zsh (default since Catalina), fallback to bash, then sh
+				if (new File("/bin/zsh").exists()) {
+					return Arrays.asList("/bin/zsh");
+				} else if (new File("/bin/bash").exists()) {
+					return Arrays.asList("/bin/bash");
+				} else {
+					return Arrays.asList("/bin/sh");
+				}
 			} else {
-				// Linux/macOS: try bash first, fallback to sh
-				String bashPath = "/bin/bash";
-				if (new File(bashPath).exists()) {
-					return Arrays.asList(bashPath);
+				// Linux and other Unix-like systems: prefer bash, fallback to sh
+				if (new File("/bin/bash").exists()) {
+					return Arrays.asList("/bin/bash");
 				} else {
 					return Arrays.asList("/bin/sh");
 				}

